.github/workflows/build_and_test.yml

```diff
@@ -60,9 +60,18 @@ jobs:
         run: |
           ./gradlew buildDockerImage
           docker run spark-kubernetes-operator:$(./gradlew properties | grep '^version:' | awk '{print $2}')
-  k8s-integration-tests:
-    name: "K8s Integration Tests"
+  k8s-integration-and-e2e-tests:
+    name: "K8s Integration & E2E Tests"
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes_versions:
+          - "1.23.0"
```

Comment:
Since Apache Spark officially supports K8s 1.27+ only, we should not have this.
 - https://github.com/apache/spark/pull/46168